src/hooks/useMenuKeyboardNavigation.ts

```diff
@@ -0,0 +1,88 @@
+import React from 'react'
+import {iterateFocusableElements} from '@primer/behaviors/utils'
+import {useMenuInitialFocus} from './useMenuInitialFocus'
+import {useTypeaheadFocus} from './useTypeaheadFocus'
+import {MenuContextProps} from '../ActionMenu'
+
+/**
+ * Keyboard navigation is a mix of 4 hooks
+ * 1. useMenuInitialFocus
+ * 2. useTypeaheadFocus
+ * 3. useCloseMenuOnTab
+ * 4. useMoveFocusToMenuItem
+ */
```

Comment:
comment: There is a lot happening in the menu, so I've tried to call the hooks from a single hook as the entry point so that it's easier to read and maintain.

There is a tiny overhead of attaching multiple event listeners. As long as we clean them up after the menu closes, the readability it provides feels like a good tradeoff